app/src/main/java/org/mozilla/tv/firefox/settings/SettingsFragment.kt

```diff
@@ -93,6 +106,61 @@ class SettingsFragment : Fragment() {
         return view
     }
 
+    private fun setupFxaProfileScreen(
+        inflater: LayoutInflater,
+        parentView: ViewGroup?
+    ): View {
+        val view = inflater.inflate(R.layout.settings_screen_fxa_profile, parentView, false)
+
+        setupFxaText(view)
+        setupFxaProfileListeners(view)
+        profileDisposable = observeFxaProfile(view)
+
+        return view
+    }
+
+    private fun setupFxaText(view: View) {
+        val appName = resources.getString(R.string.app_name)
+        view.button_firefox_tabs.text = resources.getString(R.string.fxa_settings_primary_button, appName)
+        // Username is positioned and styled differently, so it is left blank here
+        // and set on another TextView
+        view.description_top.text = resources.getString(R.string.fxa_settings_body, "")
+    }
+
+    private fun setupFxaProfileListeners(view: View) {
+        val serviceLocator = serviceLocator!!
+        val screenController = serviceLocator.screenController
+        val fxaRepo = serviceLocator.fxaRepo
+        val telemetryIntegration = TelemetryIntegration.INSTANCE
+
+        view.button_firefox_tabs.setOnClickListener {
+            // TODO show send tab tutorial
+            telemetryIntegration.fxaProfileGetTabsButtonClickEvent()
+        }
+        view.button_sign_out.setOnClickListener {
+            fxaRepo.logout()
+            screenController.handleBack(fragmentManager!!)
+            telemetryIntegration.fxaProfileSignOutButtonClickEvent()
+        }
+        view.back_button.setOnClickListener {
+            screenController.handleBack(fragmentManager!!)
```

Comment:
The web content is displayed briefly before the Nav Overlay is shown and it looks a little off: can you file a follow-up bug so we can prioritize that work?

Reply:
Already filed https://github.com/mozilla-mobile/firefox-tv/issues/2706